libobs/obs.c

```diff
@@ -2511,6 +2535,21 @@ obs_data_array_t *obs_save_sources(void)
 	return obs_save_sources_filtered(save_source_filter, NULL);
 }
 
+void obs_reset_source_uuids()
+{
+	pthread_mutex_lock(&obs->data.sources_mutex);
+
+	struct obs_source *source = obs->data.first_source;
+	while (source) {
+		bfree((void *)source->context.uuid);
+		source->context.uuid = os_get_uuid();
+
+		source = (void *)source->context.next;
+	}
+
+	pthread_mutex_unlock(&obs->data.sources_mutex);
+}
+
```

Comment:
This also assigns uuids to scenes, filters, transitions and groups which seems unintended given the "Only automatically generate UUIDs for sources" comment. Perhaps using `obs_enum_sources` with a callback would be safer?

Reply:
Scenes, filters, etc. are also sources (`OBS_OBJ_TYPE_SOURCE`), and have UUIDs as a result, which is intentional.
Perhaps the commit title could be reworded to more clearly differentiate between what we expose to the user as "sources" and the internal `struct obs_source` objects which is what it actually refers to.

Reply:
Ah I'd mixed up `OBS_OBJ_TYPE_SOURCE` and `OBS_SOURCE_TYPE_INPUT`. This is fine then, though also has the same `void *` pattern.